### ISSUE TITLE
Crash on stats generation at end

### ISSUE THREAD

**Issue by Psy-Fer** [fenderglass/ABruijn#204]
Hello,

Another organism, another assembly. I am using the same version I am with my other ticket trying to get through the repeat step (which is still running at the time of writing this). However, I have another organism i'm assembling, that got through everything, but failed at the last steps.

Here is the tail of sys.stderr:
```
[2020-01-19 21:52:35] INFO: Correcting bubbles
0% 10% 20% 30% 40% 50% 60% 70% 80% 90% 100% 
[2020-01-19 22:23:57] INFO: >>>STAGE: finalize
Traceback (most recent call last):
  File "/home/jamfer/work/Flye/bin/flye", line 25, in <module>
    sys.exit(main())
  File "/home/jamfer/work/Flye/flye/main.py", line 767, in main
    _run(args)
  File "/home/jamfer/work/Flye/flye/main.py", line 566, in _run
    jobs[i].run()
  File "/home/jamfer/work/Flye/flye/main.py", line 287, in run
    self.out_files["stats"])
  File "/home/jamfer/work/Flye/flye/assembly/scaffolder.py", line 121, in generate_stats
    cov_rate = float(tokens[2]) / float(contigs_stats[tokens[0]].coverage)
ZeroDivisionError: float division by zero
```

here is the tail of flye.log

```
[2020-01-19 12:45:57] root: INFO: >>>STAGE: polishing
[2020-01-19 12:45:57] root: INFO: Polishing genome (1/2)
[2020-01-19 12:45:59] root: INFO: Running minimap2
[2020-01-19 16:23:03] root: INFO: Separating alignment into bubbles
[2020-01-19 17:03:56] root: DEBUG: Generated 7570266 bubbles
[2020-01-19 17:03:56] root: DEBUG: Split 3007 long bubbles
[2020-01-19 17:03:56] root: DEBUG: Skipped 690 empty bubbles
[2020-01-19 17:03:56] root: DEBUG: Skipped 59 bubbles with long branches
[2020-01-19 17:03:56] root: INFO: Alignment error rate: 0.138802
[2020-01-19 17:03:56] root: INFO: Correcting bubbles
[2020-01-19 17:33:08] root: INFO: Polishing genome (2/2)
[2020-01-19 17:33:17] root: INFO: Running minimap2
[2020-01-19 21:10:07] root: INFO: Separating alignment into bubbles
[2020-01-19 21:52:34] root: DEBUG: Generated 8472601 bubbles
[2020-01-19 21:52:34] root: DEBUG: Split 3082 long bubbles
[2020-01-19 21:52:34] root: DEBUG: Skipped 1341 empty bubbles
[2020-01-19 21:52:34] root: DEBUG: Skipped 13 bubbles with long branches
[2020-01-19 21:52:35] root: INFO: Alignment error rate: 0.120760
[2020-01-19 21:52:35] root: INFO: Correcting bubbles
[2020-01-19 22:15:01] root: DEBUG: Mean contig coverage: 237, selected threshold: 47
[2020-01-19 22:15:02] root: DEBUG: Filtered 4550 contigs of total length 23880629
[2020-01-19 22:15:03] root: DEBUG: Generating polished GFA
[2020-01-19 22:23:57] root: DEBUG: 3761 sequences remained unpolished
[2020-01-19 22:23:57] root: INFO: >>>STAGE: finalize
[2020-01-19 22:25:44] root: DEBUG: ---Output dir contents:----
[2020-01-19 22:25:44] root: DEBUG: asm/
[2020-01-19 22:25:44] root: DEBUG:     88.0 M      flye.log
[2020-01-19 22:25:44] root: DEBUG:     112.0 B     params.json
[2020-01-19 22:25:44] root: DEBUG:     4.0 M       assembly_graph.gv
[2020-01-19 22:25:44] root: DEBUG:     27.0 G      assembly_graph.gfa
[2020-01-19 22:25:44] root: DEBUG:     133.0 M     assembly.fasta
[2020-01-19 22:25:44] root: DEBUG:     00-assembly/
[2020-01-19 22:25:44] root: DEBUG:         636.0 M     draft_assembly.fasta
[2020-01-19 22:25:44] root: DEBUG:     10-consensus/
[2020-01-19 22:25:44] root: DEBUG:         636.0 M     consensus.fasta
[2020-01-19 22:25:44] root: DEBUG:     20-repeat/
[2020-01-19 22:25:44] root: DEBUG:         5.0 M       graph_before_simplification.gv
[2020-01-19 22:25:44] root: DEBUG:         7.0 M       graph_before_rr.gv
[2020-01-19 22:25:44] root: DEBUG:         157.0 M     graph_before_rr.fasta
[2020-01-19 22:25:44] root: DEBUG:         5.0 M       graph_after_rr.gv
[2020-01-19 22:25:44] root: DEBUG:         97.0 M      repeat_graph_dump
[2020-01-19 22:25:45] root: DEBUG:         12.0 G      read_alignment_dump
[2020-01-19 22:25:45] root: DEBUG:         633.0 M     repeat_graph_edges.fasta
[2020-01-19 22:25:45] root: DEBUG:     21-trestle/
[2020-01-19 22:25:45] root: DEBUG:         198.0 K     repeats_dump
[2020-01-19 22:25:45] root: DEBUG:         1.0 K       trestle_summary.txt
[2020-01-19 22:25:45] root: DEBUG:         18.0 K      resolved_copies.fasta
[2020-01-19 22:25:45] root: DEBUG:         97.0 M      repeat_graph_dump
[2020-01-19 22:25:45] root: DEBUG:         636.0 M     repeat_graph_edges.fasta
[2020-01-19 22:25:45] root: DEBUG:     30-contigger/
[2020-01-19 22:25:45] root: DEBUG:         152.0 M     contigs.fasta
[2020-01-19 22:25:45] root: DEBUG:         871.0 K     contigs_stats.txt
[2020-01-19 22:25:45] root: DEBUG:         923.0 B     scaffolds_links.txt
[2020-01-19 22:25:45] root: DEBUG:         4.0 M       graph_final.gv
[2020-01-19 22:25:45] root: DEBUG:         140.0 M     graph_final.fasta
[2020-01-19 22:25:45] root: DEBUG:         27.0 G      graph_final.gfa
[2020-01-19 22:25:45] root: DEBUG:     40-polishing/
[2020-01-19 22:25:45] root: DEBUG:         156.0 M     polished_1.fasta
[2020-01-19 22:25:45] root: DEBUG:         384.0 K     contigs_stats.txt
[2020-01-19 22:25:45] root: DEBUG:         133.0 M     filtered_contigs.fasta
[2020-01-19 22:25:45] root: DEBUG:         294.0 K     filtered_stats.txt
[2020-01-19 22:25:45] root: DEBUG:         27.0 G      polished_edges.gfa
[2020-01-19 22:25:45] root: DEBUG: --------------------------
```

Something something divide by zero?

Maybe just a try block in case one of the stats was off.

Cheers.

**Comment by Psy-Fer** [comment]
Oh, one question. There is nothing extra done to the assembly at this point right? ie, I can move on with my fasta file?

**Comment by mikolmogorov** [comment]
Yep, seems like a corner case that I would need to fix! Should be easy though.

`filtered_contigs.fasta` is the almost final assembly, pending scaffolding (that might add a couple contig connections, but usually the change is not big). So feel free to use this one, if you don't want to wait.

**Comment by Psy-Fer** [comment]
Thanks.
I'm just running it again, with a full 144 core node (though I think it can only use 124 of them), just to be sure.
I'll let you know if it was just some oddity, or if it's reproducible.

**Comment by mikolmogorov** [comment]
Hmm, turns out I have already fixed this bug in 10ef30b (which was after 2.6 release). I'm assuming you were using 2.6 at this time, according to the crash log. So this should not be the problem in the latest github code anymore.
